### DISCUSSION TITLE
xterm vulnerabilities in theia/terminal

### DISCUSSION THREAD

**Discussion by BarryDrez** [eclipse-theia/theia#12689] [General]
Our security scan shows a [high vulnerability](https://nvd.nist.gov/vuln/detail/CVE-2019-0542) in theia/terminal@1.37.0.  It uses ^4.16.0.  Would there be any chance that it can use 5.0.0?  I tried using a resolution, but it doesn't help.  Thanks

**Comment by vince-fugnitto** [comment]
@BarryDrez I do not see any advisory for `xterm` for our version when performing an audit, can you confirm which version of xterm you are effectively using, if needed you may need to perform a `yarn upgrade` to update your lockfile. If you are interested you can also contribute a pull-request which bumps our xterm to the latest version (which is always welcome since it may contain improvements and bug fixes).

**Reply by BarryDrez** [reply]
Hi @vince-fugnitto.  I am using Theia 1.37 which has the terminal package.json referencing "xterm": "^4.16.0".  I have confirmed that I am using xterm 4.19.0.  According to: https://nvd.nist.gov/vuln/detail/CVE-2019-0542 , it looks like version 5.0.0 and higher are needed.  I have never done a PR.  I don't know which branches to apply this to.  I only know I will need it in 1.37 and higher.  Thank you!

**Reply by vince-fugnitto** [reply]
@BarryDrez I do not see the same advisory on either snyk or github advisories which is confusing, in any case you only need to open a pull-request and contribute the fix, no need to worry about which branch to apply the fix for.

For downstream applications the use of a [`yarn resolution`](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/) would be sufficient to pull a newer version:

```json
"resolutions": {
  "**/xterm": "^5.0.0"
}
```

**Reply by msujew** [reply]
Note that this only works in case there are no breaking changes between the xterm version that Theia uses and 5.0. Otherwise adopters will get compile/runtime errors. Right?

**Reply by BarryDrez** [reply]
@msujew Was this question for me or @vince-fugnitto ?

**Reply by msujew** [reply]
It was directed at Vince, since his proposed solution only works under specific circumstances. As seen in https://github.com/eclipse-theia/theia/pull/12691, xterm 5.0 is unfortunately not a drop-in update.
